pkg/kubelet/kubelet.go

```diff
@@ -2786,6 +2786,17 @@ func (kl *Kubelet) setNodeAddress(node *api.Node) error {
 	return nil
 }
 
+func (kl *Kubelet) updateCloudProviderFromMachineInfo(
+	node *api.Node,
+	info *cadvisorapi.MachineInfo) {
+
+	if info.CloudProvider != cadvisorapi.UnkownProvider &&
+		info.CloudProvider != cadvisorapi.Baremetal {
+		node.Spec.ProviderID = strings.ToLower(string(info.CloudProvider)) +
+			":////" + string(info.InstanceID)
```

Comment:
Sorry, I probably should have documented it. 
If info.CloudProvider will always have a non empty value (it has a set of defined values in info/v1/machine.go. If it is not UnkownProvider (typo in code, will fix here) nor Baremetal then it is the name of a real cloud service. Same thing goes for info.InstanceID , it will be receive its value from a function that is supposed to return "Unknown" if there was an error while retrieving the value.
":////" is to conform with kubernetes's cloudproviders (pkg/cloudprovider/providers/*) that will write to ProviderID if cloud-provider flag is given to kubelet. The most verbose one write in the form of "cloudprovider://project/availability_zone/instance_name". Here I dont have project name or availability zone so I return them empty.